test/integration/rt_channel_test.exs

```diff
@@ -263,6 +264,55 @@ defmodule Realtime.Integration.RtChannelTest do
       }
     end
 
+    @tag policies: [
+           :authenticated_read_broadcast_and_presence,
+           :authenticated_write_broadcast_and_presence
+         ],
+         topic: ":topic"
+    test "private broadcast with valid channel a colon character sends message and won't intercept in public channels",
+         %{topic: topic} do
+      {anon_socket, _} = get_connection("anon")
+      {socket, _} = get_connection("authenticated")
+      topic = "realtime:#{topic}"
+
+      WebsocketClient.join(socket, topic, %{
+        config: %{broadcast: %{self: true}, private: true}
+      })
+
+      WebsocketClient.join(anon_socket, topic, %{
```

Comment:
@filipecabaco the topic for `private: false` case should end up being `realtime:private:topic` b/c that's the loophole that this PR patches.

Reply:
Should be just `private:topic`

Reply:
made the change in the test 👍